core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -151,208 +165,182 @@ class TransactionsTest extends KafkaServerTestHarness {
         }
       }
     } finally {
-      producer.close()
       consumer.close()
     }
 
-    // Inspite of random aborts, we should still have exactly 1000 messages in topic2. Ie. we should not
+    // In spite of random aborts, we should still have exactly 1000 messages in topic2. I.e. we should not
     // re-copy or miss any messages from topic1, since the consumed offsets were committed transactionally.
-    val verifyingConsumer = transactionalConsumer("foobargroup")
+    val verifyingConsumer = transactionalConsumers.head
     verifyingConsumer.subscribe(List(topic2))
     val valueSeq = TestUtils.pollUntilAtLeastNumRecords(verifyingConsumer, numSeedMessages).map { record =>
       TestUtils.assertCommittedAndGetValue(record).toInt
     }
-    verifyingConsumer.close()
     val valueSet = valueSeq.toSet
     assertEquals(s"Expected $numSeedMessages values in $topic2.", numSeedMessages, valueSeq.size)
     assertEquals(s"Expected ${valueSeq.size} unique messages in $topic2.", valueSeq.size, valueSet.size)
   }
 
   @Test
   def testFencingOnCommit() = {
-    val transactionalId = "my-t.id"
-    val producer1 = TestUtils.createTransactionalProducer(transactionalId, servers)
-    val producer2 = TestUtils.createTransactionalProducer(transactionalId, servers)
-    val consumer = transactionalConsumer()
+    val producer1 = transactionalProducers.head
+    val producer2 = transactionalProducers.tail.head
```

Comment:
Thanks for reviewing. We could retrieve by index too. I'll change `.tail.head` to `.get(1)`. I hope that works.